component/attr_cache/attr_cache.go

```diff
@@ -262,9 +271,7 @@ func (ac *AttrCache) StreamDir(options internal.StreamDirOptions) ([]*internal.O
 	log.Trace("AttrCache::ReadDir : %s", options.Name)
 
 	pathList, token, err := ac.NextComponent().StreamDir(options)
-	if err == nil && options.Offset < maxFilesPerDir {
-		ac.cacheAttributes(pathList)
-	}
+	ac.cacheAttributes(pathList)
```

Comment:
shouldn't we put the max files check here ?

Reply:
the was there in the cacheAttributes method so i thought we could skip her, but we can avoid iterating over path list if do the check here so added it back

Reply:
I revisited the code, we can remove this check from here and just keep err=nil check as the ac.cacheAttributes() method is anyway checking on the length of list against the new configured parameter.